### PR TITLE
Lift state from FacetsGrid to App

### PR DESCRIPTION
**Pull request by alahwa**
The end result is that App becomes the only stateful container and the FacetResponse counts correctly propagate to the Header. Note that as we continue to add logic, we might want to consider want a logic-containing component to reside under App that holds all of the FacetResponse-dependent components within it.

### CODE REVIEW

File: ui/src/App.js [hunk 1]
```diff
@@ -27,33 +45,23 @@ class App extends Component {
...
                     <Header
                         datasetName={this.state.datasetName}
-                        totalCount={this.state.facetsResponse.count}
+                        totalCount={this.state.totalCount}
                     />
-                    <FacetsGrid facetsResponse={this.state.facetsResponse} />
+                    <FacetsGrid
+                        updateFacets={this.updateFacets}
+                        facets={this.state.facets}
+                        totalCount={this.state.totalCount}/>
```
Comment (melissachang) [inline ui/src/App.js]: From https://github.com/airbnb/javascript/tree/master/react#alignment
```
+                    <FacetsGrid
￼ +                        updateFacets={this.updateFacets}
￼ +                        facets={this.state.facets}
￼ +                        totalCount={this.state.totalCount}
                       />
```

File: ui/src/components/FacetsGrid.js [hunk 2]
```diff
@@ -1,102 +1,32 @@
...
+    const facets = props.facets;
+    const totalCount = props.totalCount;
+    const updateFacets = props.updateFacets;
+    const facetsList = facets.map((facet) =>
+        <GridTile key={facet.name}>
+            <FacetCard
+                facet={facet}
+                count={totalCount}
+                updateFacets={updateFacets}
+                className='facetCard'/>
```
Comment (melissachang) [inline ui/src/components/FacetsGrid.js]: `/>` on next line

File: ui/src/App.js (hunk 1 above)
Comment (alahwa) [reply ui/src/App.js]: Done

File: ui/src/components/FacetsGrid.js (hunk 2 above)
Comment (alahwa) [reply ui/src/components/FacetsGrid.js]: Done

**Pull request merged**